File: 2022/day4/b.ts
const input = await Deno.readTextFile("input.txt");
const pairs = input.trim().split(/\r?\n/);

function main() {
  let overlapCount = 0;
  pairs.map((s) => {
    const sections = s.split(/,|-/);
    const [a, b, x, y] = sections.map(Number);
    if ((b >= x && a <= y)|| (y >= a && x <= b)){
      overlapCount++;
    }
  });
  console.log("overlap count: ", overlapCount);
}

main();

File: 2022/day4/b.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer file.Close()

	sc := bufio.NewScanner(file)
	overlapCount := 0
	for sc.Scan() {
		var a, b, x, y int
		fmt.Sscanf(sc.Text(), "%d-%d,%d-%d", &a, &b, &x, &y)
		if (b >= x && a <= y) || (y >= a && x <= b) {
			overlapCount++
		}
	}
	fmt.Printf("count: %d\n", overlapCount)
}
